src/samplereader/FragmentedSampleReader.cpp

```diff
@@ -425,12 +425,15 @@ AP4_Result CFragmentedSampleReader::ProcessMoof(AP4_ContainerAtom* moof,
 
       // Inform decrypter of pattern decryption (CBCS)
       AP4_UI32 schemeType = m_protectedDesc->GetSchemeType();
-      if (schemeType == AP4_PROTECTION_SCHEME_TYPE_CENC || schemeType == AP4_PROTECTION_SCHEME_TYPE_CBCS)
+      if (schemeType == AP4_PROTECTION_SCHEME_TYPE_CENC ||
+          schemeType == AP4_PROTECTION_SCHEME_TYPE_PIFF ||
+          schemeType == AP4_PROTECTION_SCHEME_TYPE_CBCS)
       {
         m_readerCryptoInfo.m_cryptBlocks = sample_table->GetCryptByteBlock();
         m_readerCryptoInfo.m_skipBlocks = sample_table->GetSkipByteBlock();
 
-        if (schemeType == AP4_PROTECTION_SCHEME_TYPE_CENC)
+        if (schemeType == AP4_PROTECTION_SCHEME_TYPE_CENC ||
+            schemeType == AP4_PROTECTION_SCHEME_TYPE_PIFF)
```

Comment:
Just out of curiosity. Viewing on my phone and notice there is no check for CBCS here vs the other places.

Reply:
its right because cbcs use AES_CBC